Shared/Supporting Files/Extensions/View+Sheet.swift

```diff
@@ -286,44 +286,45 @@ private struct Sheet<Content>: UIViewRepresentable where Content: View {
         /// A Boolean value indicating whether the presented view controller is a hosting controller.
         let isPresentedControllerHostingType = rootViewController.presentedViewController is UIHostingController<Content>
         
+        /// A Boolean value indicating whether the presented view controller is an alert controller.
+        let isPresentedControllerAlertType = rootViewController.presentedViewController is UIAlertController
+        
+        /// A Boolean value indicating whether the sheet was already presenting.
+        let wasPresenting = rootViewController.presentedViewController != nil && isPresentedControllerHostingType
+        
+        /// A Boolean value indicating whether the hosting controller is being dismissed.
+        let isHostBeingDismissed = model.hostingController.isBeingDismissed
+        
         // Ensures that the device's layout is such that a sheet should be presented and
         // the hosting controller's sheet presentation controller exists.
         guard isSheetLayout,
               let sheet = model.hostingController.sheetPresentationController else {
             // Dismisses the sheet if it is being presented and a popover should
             // be presented instead.
-            if isPresentedControllerHostingType && !model.hostingController.isBeingDismissed {
+            if isPresentedControllerHostingType && !isHostBeingDismissed {
                 rootViewController.dismiss(animated: false)
             }
             return
         }
         
-        /// A Boolean value indicating whether the sheet was already presenting.
-        let wasPresenting = rootViewController.presentedViewController != nil
-        
-        /// A Boolean value indicating whether the hosting controller is being dismissed.
-        let isHostBeingDismissed = model.hostingController.isBeingDismissed
-        
-        /// A Boolean value indicating whether the presented view controller is an alert controller.
-        let isPresentedControllerAlertType = rootViewController.presentedViewController is UIAlertController
-        
         if isPresented && !wasPresenting {
             // Sets the sheet presentation controller's delegate.
             sheet.delegate = context.coordinator
             // Configures the hosting controller's sheet presentation controller.
             configureSheetPresentationController(sheet)
             // Presents the hosting controller.
-            rootViewController.present(model.hostingController, animated: model.isTransitioningFromPopover ? false : true)
+            rootViewController.present(model.hostingController, animated: !model.isTransitioningFromPopover)
         } else if !isPresented && wasPresenting && !isHostBeingDismissed && !isPresentedControllerAlertType {
```

Comment:
The changes solved most of the problems, but there is still an issue when an alert and sheet are in conflict. Run the test code on an iPad and add an alert button
```swift
Button("alert") {
    isShowingAlert = true
}
.alert(isPresented: $isShowingAlert) {
    Alert(title: Text("Alert!"))
}
```
When I tap either one of the sheet buttons, then tap the (grayed) alert button while the sheet is presenting, in console it throws the error

> Attempt to present *AlertController* on *UIHostingController: 0x111017200* (from <SwiftUI.UIKitNavigationController: 0x11782e400>) which is already presenting

I think there is still room for improvement to these conditions and I'll try to sort it out. 🧐

Reply:
Wouldn't this work in that case? @yo1995 
```swift
let wasPresenting = rootViewController.presentedViewController != nil && isPresentedControllerHostingType && !isPresentedControllerAlertType
```

Reply:
No it won't. I think the problem is that on iPad, the [formSheet](https://developer.apple.com/documentation/uikit/uimodalpresentationstyle/formsheet) styled sheet is dismissed too late. So when the alert is presented, the sheet is still on the screen, which causes the error.

I thought I found a simple fix, but it was a red herring 🥲 . I'll keep looking for a little while, but it shouldn't hold up the review of this PR.

Reply:
This seems like it would require a solution similar to the timing bug fix, but I am not sure how I would go about implementing that. Should I just note this limitation as part of the documentation?